### ISSUE TITLE
🐛 [APP_CHECK] Instructions for activating the debug provider on IOS don't appear to work

### ISSUE THREAD

**Issue by PaulRudin** [firebase/flutterfire#6872]
## Bug report

I have followed the instructions for activating the debug Appcheck provider here: https://firebase.flutter.dev/docs/app-check/usage/#activating-the-debug-provider-iosmacos but do not see a logged debug token.

App check works fine running on real ios devices, but when I attempt to run on a simulator, I don't get a debug token logged, and I can see in the firebase console that attempts to read the realtime database are being blocked by appcheck.

### Steps to reproduce

1. Modify ios/Runner/AppDeletage.swift. If I used exactly the version from the instructions I get build errors. This is the version I'm trying:
```swift
import UIKit
import Flutter

import google_mobile_ads

import Firebase

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {

  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    GeneratedPluginRegistrant.register(with: self)
      
    let listTileFactory = ListTileNativeAdFactory()
    FLTGoogleMobileAdsPlugin.registerNativeAdFactory(
        self, factoryId: "listTile", nativeAdFactory: listTileFactory)

    let providerFactory = AppCheckDebugProviderFactory()
    AppCheck.setAppCheckProviderFactory(providerFactory)

      
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}
```

2. Ensure that your app uses appcheck protected services and that appcheck is enforced.
3. Run the app on a simulator with e.g. `flutter run`
4. Check the logs - there's no debug token printed.

### Expected behavior

I expect a debug appcheck token to be logged.

## Additional context

Appcheck works on real ios devices for the same app.

---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.2 20G95 darwin-arm, locale en-GB)
[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.

[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[!] Android Studio (not installed)
[✓] Connected device (3 available)

! Doctor found issues in 2 categories.
```

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Changing current working directory to: /Users/paulrudin/Dropbox/livefpl-ui
Dart SDK 2.13.4
Flutter SDK 2.2.3
liveff 0.0.116+64

dependencies:
- admob_consent 2.0.0 [flutter]
- cloud_firestore 2.4.0 [cloud_firestore_platform_interface cloud_firestore_web firebase_core firebase_core_platform_interface flutter meta]
- cupertino_icons 1.0.3
- firebase 9.0.1 [http http_parser js]
- firebase_analytics 8.2.0 [firebase_analytics_platform_interface firebase_analytics_web firebase_core flutter meta]
- firebase_app_check 0.0.1+3 [firebase_app_check_platform_interface firebase_app_check_web firebase_core firebase_core_platform_interface flutter]
- firebase_auth 3.0.1 [firebase_auth_platform_interface firebase_auth_web firebase_core firebase_core_platform_interface flutter meta]
- firebase_auth_web 3.0.0 [firebase_auth_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins http_parser intl js meta]
- firebase_core 1.5.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_database 7.2.0 [firebase_core flutter]
- fluro 2.0.3 [flutter]
- flutter 0.0.0 [characters collection meta typed_data vector_math sky_engine]
- flutter_riverpod 1.0.0-dev.6 [collection flutter meta riverpod state_notifier]
- flutter_staggered_grid_view 0.4.0 [flutter]
- google_mobile_ads 0.13.3 [meta flutter]
- http 0.13.3 [async http_parser meta path pedantic]
- is_firebase_test_lab_activated 2.0.0 [flutter]
- logging 1.0.1
- rxdart 0.27.1
- shared_preferences 2.0.6 [meta flutter shared_preferences_platform_interface shared_preferences_linux shared_preferences_macos shared_preferences_web shared_preferences_windows]
- stream_transform 2.0.0
- universal_platform 1.0.0+1
- uuid 3.0.4 [crypto]

dev dependencies:
- build_runner 2.1.1 [args async analyzer build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pedantic pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver archive async boolean_selector characters charcode clock collection crypto matcher platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math]
- flutter_launcher_icons 0.9.1 [args image path yaml]
- flutter_test 0.0.0 [flutter test_api path fake_async clock stack_trace vector_math async boolean_selector characters charcode collection matcher meta source_span stream_channel string_scanner term_glyph typed_data]
- integration_test 0.0.0 [flutter flutter_driver flutter_test path vm_service archive async boolean_selector characters charcode clock collection crypto fake_async file matcher meta source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api typed_data vector_math webdriver]
- mockito 5.0.14 [analyzer build code_builder collection dart_style matcher meta path source_gen test_api]
- test 1.16.8 [analyzer async boolean_selector collection coverage http_multi_server io js node_preamble package_config path pedantic pool shelf shelf_packages_handler shelf_static shelf_web_socket source_span stack_trace stream_channel typed_data web_socket_channel webkit_inspection_protocol yaml test_api test_core]

transitive dependencies:
- _fe_analyzer_shared 22.0.0 [meta]
- analyzer 1.7.1 [_fe_analyzer_shared cli_util collection convert crypto glob meta package_config path pub_semver source_span watcher yaml pedantic]
- archive 3.1.2 [crypto path]
- args 2.2.0
- async 2.6.1 [meta collection]
- boolean_selector 2.1.0 [source_span string_scanner]
- build 2.1.0 [analyzer async convert crypto glob logging meta path]
- build_config 1.0.0 [checked_yaml json_annotation path pubspec_parse yaml]
- build_daemon 3.0.0 [built_collection built_value http_multi_server logging pedantic path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.0.4 [analyzer async build crypto graphs logging path package_config pool pub_semver stream_transform yaml]
- build_runner_core 7.1.0 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta path package_config pedantic pool timing watcher yaml]
- built_collection 5.1.0
- built_value 8.1.2 [built_collection collection fixnum meta]
- characters 1.1.0
- charcode 1.2.0
- checked_yaml 2.0.1 [json_annotation source_span yaml]
- cli_util 0.3.3 [meta path]
- clock 1.1.0
- cloud_firestore_platform_interface 5.3.0 [collection firebase_core flutter meta plugin_platform_interface]
- cloud_firestore_web 2.3.0 [cloud_firestore_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- code_builder 4.1.0 [built_collection built_value collection matcher meta]
- collection 1.15.0
- convert 3.0.1 [typed_data]
- coverage 1.0.3 [args logging package_config path source_maps stack_trace vm_service]
- crypto 3.0.1 [collection typed_data]
- dart_style 2.0.3 [analyzer args path pub_semver source_span]
- fake_async 1.2.0 [clock collection]
- ffi 1.1.2
- file 6.1.0 [meta path]
- firebase_analytics_platform_interface 2.0.1 [flutter meta]
- firebase_analytics_web 0.3.0+1 [firebase firebase_analytics_platform_interface flutter flutter_web_plugins meta]
- firebase_app_check_platform_interface 0.0.1+3 [firebase_core flutter meta plugin_platform_interface]
- firebase_app_check_web 0.0.1+3 [firebase_app_check_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins js]
- firebase_auth_platform_interface 6.0.0 [firebase_core flutter meta plugin_platform_interface]
- firebase_core_platform_interface 4.0.1 [collection flutter meta plugin_platform_interface]
- firebase_core_web 1.1.0 [firebase_core_platform_interface flutter flutter_web_plugins js meta]
- fixnum 1.0.0
- flutter_web_plugins 0.0.0 [flutter js characters collection meta typed_data vector_math]
- freezed_annotation 0.14.2 [collection json_annotation meta]
- frontend_server_client 2.1.0 [async path]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- glob 2.0.1 [async collection file path pedantic string_scanner]
- graphs 2.0.0
- http_multi_server 3.0.1 [async]
- http_parser 4.0.0 [charcode collection source_span string_scanner typed_data]
- image 3.0.2 [archive meta xml]
- intl 0.17.0 [clock path]
- io 1.0.3 [meta path string_scanner]
- js 0.6.3
- json_annotation 4.0.1
- matcher 0.12.10 [stack_trace]
- meta 1.3.0
- mime 1.0.0
- node_preamble 2.0.1
- package_config 2.0.0 [path]
- path 1.8.0
- path_provider_linux 2.0.2 [flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.0.1 [flutter meta platform plugin_platform_interface]
- path_provider_windows 2.0.3 [ffi flutter meta path path_provider_platform_interface win32]
- pedantic 1.11.1
- petitparser 4.1.0 [meta]
- platform 3.0.0
- plugin_platform_interface 2.0.1 [meta]
- pool 1.5.0 [async stack_trace]
- process 4.2.1 [file path platform]
- pub_semver 2.0.0 [collection]
- pubspec_parse 1.0.0 [checked_yaml collection json_annotation pub_semver yaml]
- riverpod 1.0.0-dev.6 [collection freezed_annotation meta state_notifier]
- shared_preferences_linux 2.0.2 [file meta flutter path path_provider_linux shared_preferences_platform_interface]
- shared_preferences_macos 2.0.2 [flutter shared_preferences_platform_interface]
- shared_preferences_platform_interface 2.0.0 [flutter]
- shared_preferences_web 2.0.1 [flutter flutter_web_plugins meta shared_preferences_platform_interface]
- shared_preferences_windows 2.0.2 [flutter file meta path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- shelf 1.2.0 [async collection http_parser path stack_trace stream_channel]
- shelf_packages_handler 3.0.0 [path shelf shelf_static]
- shelf_static 1.1.0 [convert http_parser mime path shelf]
- shelf_web_socket 1.0.1 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.0.3 [analyzer async build dart_style glob meta path pedantic source_span yaml]
- source_map_stack_trace 2.1.0 [path stack_trace source_maps]
- source_maps 0.10.10 [source_span]
- source_span 1.8.1 [collection path term_glyph]
- stack_trace 1.10.0 [path]
- state_notifier 0.7.0 [meta]
- stream_channel 2.1.0 [async]
- string_scanner 1.1.0 [charcode source_span]
- sync_http 0.3.0
- term_glyph 1.2.0
- test_api 0.3.0 [async boolean_selector collection meta path source_span stack_trace stream_channel string_scanner term_glyph matcher]
- test_core 0.3.19 [analyzer async args boolean_selector collection coverage glob io meta package_config path pedantic pool source_map_stack_trace source_maps source_span stack_trace stream_channel vm_service yaml matcher test_api]
- timing 1.0.0 [json_annotation]
- typed_data 1.3.0 [collection]
- vector_math 2.1.0
- vm_service 6.2.0
- watcher 1.0.0 [async path pedantic]
- web_socket_channel 2.1.0 [async crypto stream_channel]
- webdriver 3.0.0 [archive matcher path stack_trace sync_http]
- webkit_inspection_protocol 1.0.0 [logging]
- win32 2.2.5 [ffi]
- xdg_directories 0.2.0 [meta path process]
- xml 5.1.2 [collection meta petitparser]
- yaml 3.1.0 [collection source_span string_scanner]

```

</details>

---

**Comment by markusaksli-nc** [comment]
Hi @PaulRudin 
Are there any errors logged? Have you tried using the latest version `firebase_app_check: ^0.0.2`?
Thank you

**Comment by PaulRudin** [comment]
I have upgraded firebase_app_check to 0.0.2, but the behaviour is the same. There's no error logged.

**Comment by markusaksli-nc** [comment]
Have you tried the instructions in the [firebase docs](https://firebase.google.com/docs/app-check/ios/debug-provider?hl=uk#simulator)?

**Comment by AresVampire** [comment]
I am facing the same issue, debug token is not generated. with the latest version 0.0.3

**Comment by moritzpail** [comment]
I ran into the same issue today running ( with `firebase_app_check: ^0.0.6+2`). 

I found that while no debug token is printed to the console when running `flutter run` the token was printed to the console when running the project directly from Xcode. I am not sure what exactly the issue is, but maybe it helps.

**Comment by rohan20** [comment]
For everyone struggling with this issue, what did you end up doing? Not enforcing AppCheck?

Or did you stop using the iOS Simulator for dev?

**Comment by PaulRudin** [comment]
> 

You don't see the output in the console from `flutter run`, but if you launch via xcode, then you can see it in the output that xcode shows.

**Comment by OfirMarom** [comment]
In case this helps anyone, I got the debug token to generate with firebase_app_check v0.0.8. As previously mentioned, the debug token displays when you launch directly from xcode.

In addition, I found that these lines:

```
let providerFactory = AppCheckDebugProviderFactory()
AppCheck.setAppCheckProviderFactory(providerFactory)

```

Must occur BEFORE this line:

`GeneratedPluginRegistrant.register(with: self)`

I have no idea why, but the other way did not generate a token. I suspect it is related to this line in the documentation: "In your debug build *before using any Firebase backend services*, create and set the App Check debug provider factory".
